docs/kb/rules/using-semgrep-rule-schema-in-vscode.md

```diff
@@ -0,0 +1,71 @@
+---
+description: Use the Semgrep rule schema in VS Code to help make rule writing easier.
+tags:
+  - Rules
+  - VS Code
+---
+
+import MoreHelp from "/src/components/MoreHelp"
+
+# Use the Semgrep rule schema to write rules in VS Code
+
+You may already be familiar with writing rules in the [Semgrep Playground](/docs/playground). However, if your IDE of choice is VS Code and you'd like to write Semgrep rules there, utilizing the Semgrep rule schema can be a great aid while doing so. The Semgrep rule schema will provide a richer editing environment, allowing for VS Code to understand the shape of your rule's YAML file, including its value sets, defaults, and descriptions ([reference](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-schemas)).
+
+:::tip
+Writing rules locally in your IDE is also helpful when iteratively testing your rule against an entire local repository, as opposed to just a snippet of test code.
+:::
+
+When setup properly, you can expect to get auto-completion options in your VS Code IDE just as you would in the Semgrep Playground when writing rules:
+
+![Example Semgrep YAML rule file with auto-complete](/img/kb/vscode-schema-autocomplete-example.png)
+
+## Adding the Semgrep rule schema in VS Code
+
+Adding the Semgrep rule schema in VS Code requires two steps:
+
+1. Install the YAML Language Support extension by Red Hat
+2. Associate the Semgrep rule schema
+
+### Install the YAML Language Support extension by Red Hat
+
+You can install the extension either directly in VS Code or by going to the Visual Studio Marketplace and installing it from there. In VS Code, typically just searching `yaml` will yield the correction extension as the top result. However, please verify you are installing the correct extension by ensuring it is the same as the one [linked here](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml).
+
+### Associate the Semgrep rule schema
+
+Once the extension is properly installed, you will need to associate the Semgrep rule schema with any Semgrep YAML rule definitions you are working on in VS Code. This can be done in several ways and we suggest first reading the [extension overview](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-schemas) to fully understand all of them. Here, we will cover two of them.
```

Comment:
I think I'd go for a different structure here, something like:

> Once the extension is installed, associate the Semgrep rule schema with Semgrep YAML rule definitions you are working on in VS Code using one of two recommended methods.

<method 1 - yaml file>
<method 2 - glob>

And then place the link to the VS Code docs as more context, like:

> Alternatively, consult the [extension overview](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-schemas) for more details and other methods.

Reply:
I can make the 2 methods clearer in the intro, but I'm hesitant to change the wording of looking at the extension overview as an "alternative". I really want the reader to look there first to start getting an idea before following the KB article

Reply:
I committed a change just now to address this.